source/common/http/http1/codec_impl.cc

```diff
@@ -459,14 +464,24 @@ Status RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool e
     upgrade_request_ = true;
   }
 
-  connection_.copyToBuffer(method->value().getStringView().data(), method->value().size());
-  connection_.addCharToBuffer(' ');
+  auto& conn_buffer_helper = connection_.bufferHelper();
+
+  const uint64_t basic_header_size = method->value().size() + (host ? host->value().size() : 0) +
```

Comment:
I put it in the another comment but it's been lost.

I think this kind of code is hard to reason about and we should offer a Buffer::add that takes an ordered collection of string_view and do this all inside Buffer::add.